r-package/grf/tests/testthat/test_regression_forest.R

```diff
@@ -213,3 +213,19 @@ test_that("inverse propensity weighting in the training of a regression forest w
   ipw.mse.forest.weighted <- sum((predict(forest.weighted, X) - Y)^2)
   expect_true(ipw.mse.forest.weighted < ipw.mse.forest)
 })
+
+test_that("a non-pruned honest regression forest has lower MSE than a pruned honest regression forests
```

Comment:
Looks good overall. A few minor things for the tests:
- It'd be good to have explicit separation in the MSE, e.g., something like expect_true(mse.notpruned < 0.6 * mse.pruned). This gives more confidence that the feature is actually helping.
- Could you please also add a test for causal_forest, just to make sure that code path in fact gets exercised during testing?